contracts/interfaces/IbcReceiver.sol

```diff
@@ -12,16 +12,11 @@ import {ChannelOrder, CounterParty, IbcPacket, AckPacket} from "../libs/Ibc.sol"
  * handshake callbacks.
  */
 interface IbcChannelReceiver {
-    function onOpenIbcChannel(
-        string calldata version,
-        ChannelOrder ordering,
-        bool feeEnabled,
-        string[] calldata connectionHops,
-        CounterParty calldata counterparty
-    ) external returns (string memory selectedVersion);
+    function onChannelOpenInit(string calldata version) external returns (string memory selectedVersion);
 
-    function onConnectIbcChannel(bytes32 channelId, bytes32 counterpartyChannelId, string calldata counterpartyVersion)
-        external;
+    function onChannelOpenTry(string calldata counterpartyVersion) external returns (string memory selectedVersion);
```

Comment:
At this point the dapp is only responsible for the version selection and it seems like providing the version and counterparty version is enough. Something like this
- initiator sets local version to X 
- initiator dapp checks that X is valid, so it becomes the "selected version"
- on remote chain, channOpenTry is called with counterpartyVersion = X
- remote dapp is called with counterparty version X and if it supports it, it becomes the selected version. 
I think this is cleaner than having to compare local with remote version

Reply:
how would the initiator dapp be notified if the remote chain doesn't support the version - I'm assuming that wouldn't be called in a handler right? So would it just timeout?

Reply:
The remote chain will error out and the channel will never be established. Channel handshake messages don't really timeout as packet messages do. The channel will stay in INIT state